src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -81,28 +80,26 @@ object DepositHandler {
 
   def finalizeDeposit(mimeType: String)(implicit settings: Settings, id: String): Try[Unit] = {
     log.info(s"[$id] Finalizing deposit")
-    // TODO pass on the combination object of baseDir and baseURL
-    implicit val baseDir: File = new File(settings.bagStoreBaseDir)
-    implicit val baseUrl: URI  = new URI(settings.bagStoreBaseUri)
+    implicit val optBagStoreSettings: Option[BagStoreSettings] = settings.bagStoreSettings
     val tempDir = new File(settings.tempDir, id)
 
     val result = for {
-      _        <- checkBagStoreBaseDir()
+      _        <- optBagStoreSettings.map(checkBagStoreBaseDir).getOrElse(Success(()))
```

Comment:
This is something different from what we discussed earlier.  Here we have both `base-dir `and `base-url` present, but the path to the `base-dir` doesn't exist. Do you really want to have this checked in `ServiceInitializer`, which means that the whole service doesn't start when the directory doesn't exist?  What if when the service is started, the `base-dir` is there, but later for some reason it gets deleted?